### PR TITLE
bpo-39783: Optimize contruction of Path and PurePath objects

### PR DESCRIPTION
**Pull request by remilapeyre**
Now that the `_closed` attribute has been removed we can move ahead with this.

https://bugs.python.org/issue39783

### CODE REVIEW

**Comment by barneygale** [comment]
Could this be implemented in `_from_parts()` to avoid doubling up the code?

**Comment by remilapeyre** [comment]
> Could this be implemented in `_from_parts()` to avoid doubling up the code?

Yes, thanks!

**Comment by barneygale** [comment]
LGTM!

File: Lib/pathlib.py [hunk 1]
```diff
@@ -686,7 +686,9 @@ def _parse_args(cls, args):
         return cls._flavour.parse_parts(parts)
 
     @classmethod
-    def _from_parts(cls, args, init=True):
+    def _from_parts(cls, args, kwargs=None, *, init=True):
```
Comment (akulakov) [inline Lib/pathlib.py:689]: It looks like `kwargs` is not needed as it's not used anywhere?

Comment (iritkatriel) [reply Lib/pathlib.py:689]: It has been removed on main, you have a merge conflict now.

**Comment by barneygale** [comment]
Closing - see rationale in issue. Thanks all the same for the patch.

**Pull request closed without merging**